### ISSUE TITLE
YAM2.8dev 10.07.2012 editing sent msg DSI

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#332]
**Originally by _javierdlr@euskalnet.net_ on 2012-07-10 18:30:42 +0200**

___
## Steps to reproduce
 1.go to SENT folder adn select msg
 2.click on EDIT button to try to edit such msg

## Expected results
editing ;-)
## Actual results
DSI/crashlog:
```
Task "YAM"
Generated by GrimReaper 53.5
Crash occured in module kernel at address 0x0141C558
Type of crash: unknown exception

Register dump:
GPR (General Purpose Registers):
   0: 00000010 6C0BE7D0 01448454 0140D6EC 81000009 6C0BE778 0141EA84 01CA2AB4 
   8: 81000000 0140D6EC 20000000 0141C544 0000015C 6BFB4324 6C0BD8BC 6C0BD8AC 
  16: 00000000 6C0F0000 00000000 6A460AE0 6C0F0000 698A58A0 FFFFFFFF 00000007 
  24: 698E372C 69D384C0 698A32C8 6BFB0000 00000006 6A6C17E4 69D386EC 69D386E8 


FPR (Floating Point Registers, NaN = Not a Number):
   0:              nan         0.359375          0.99999                0 
   4:                0             16.8            127.8               64 
   8:               64       4.5036e+15       4.5036e+15               10 
  12:      2.36963e-14              0.5                0                0 
  16:                0                0                0                0 
  20:                0                0                0                0 
  24:                0                0                0                0 
  28:                0                0                0                0 

FPSCR (Floating Point Status and Control Register): 0x82024000


SPRs (Special Purpose Registers):
           Machine State (msr) : 0x0200F030
                Condition (cr) : 0x28822084
      Instruction Pointer (ip) : 0x0141C558
       Xtended Exception (xer) : 0x20000000
                   Count (ctr) : 0x0141C544
                     Link (lr) : 0x017BC310
            DSI Status (dsisr) : 0x00017C00
            Data Address (dar) : 0x800017E0



680x0 emulated registers:
DATA: 6C09B1FE 00000003 00000000 00000000 00000000 00000000 00000000 00000000 
ADDR: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 6C0BFF10 
FPU0:                0                0                0                0 
FPU4:                0                0                0                0 



Symbol info:
Instruction pointer 0x0141C558 belongs to module "kernel" (HUNK/Kickstart)

Stack trace:
    native kernel module kernel+0x0001c558
    native kernel module newlib.library.kmod+0x00005b68
    YAM:NewEditMailWindow()+0x4d4 (section 1 @ 0x844E8)
    YAM:NewMessage()+0xf4 (section 1 @ 0x5C9CC)
    YAM:MA_NewMessageFunc()+0x24 (section 1 @ 0x5CD94)
    native kernel module kernel+0x00047ce4
    muimaster.library:NotifyDispatcher()+0x3b4 (section 1 @ 0x5CE40)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x000082dc
    muimaster.library:ApplicationDispatcher()+0x440 (section 1 @ 0x27BE0)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x000082dc
    YAM:YAMApplicationDispatcher()+0x124 (section 1 @ 0xA96F0)
    muimaster.library:CustomClassDispatcher()+0xa0 (section 1 @ 0x417C)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x00008424
    TheButton.mcc:mSendNotify.clone.3()+0x18c (section 1 @ 0x6670)
    TheButton.mcc:_Dispatcher()+0xda4 (section 1 @ 0x7470)
    muimaster.library:CustomClassDispatcher()+0xa0 (section 1 @ 0x417C)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x00008424
    muimaster.library:m_Notify_SET()+0x380 (section 1 @ 0x5C5A4)
    muimaster.library:m_Area_SET()+0x250 (section 1 @ 0x2F6E4)
    muimaster.library:AreaDispatcher()+0x4f0 (section 1 @ 0x308BC)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x000082dc
    TheButton.mcc:mSets()+0x248 (section 1 @ 0x56A0)
    TheButton.mcc:_Dispatcher()+0x750 (section 1 @ 0x6E1C)
    muimaster.library:CustomClassDispatcher()+0xa0 (section 1 @ 0x417C)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x0001870c
    native kernel module intuition.library.kmod+0x00007f14
    native kernel module intuition.library.kmod+0x000081b4
    muimaster.library:m_Area_HandleEvent()+0x7d4 (section 1 @ 0x2E038)
    muimaster.library:AreaDispatcher()+0xaa0 (section 1 @ 0x30E6C)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x000082dc
    TheButton.mcc:_Dispatcher()+0xe0 (section 1 @ 0x67AC)
    muimaster.library:CustomClassDispatcher()+0xa0 (section 1 @ 0x417C)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x00008308
    native kernel module intuition.library.kmod+0x00007fbc
    muimaster.library:SendIMsgs()+0x360 (section 1 @ 0x74A98)
    muimaster.library:m_Window_Input()+0x878 (section 1 @ 0x78CF0)
    muimaster.library:m_Application_NewInput()+0x7b0 (section 1 @ 0x24008)
    muimaster.library:ApplicationDispatcher()+0x544 (section 1 @ 0x27CE4)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x000082dc
    YAM:YAMApplicationDispatcher()+0x124 (section 1 @ 0xA96F0)
    muimaster.library:CustomClassDispatcher()+0xa0 (section 1 @ 0x417C)
    native kernel module intuition.library.kmod+0x00018178
    native kernel module intuition.library.kmod+0x0001839c
    native kernel module intuition.library.kmod+0x00008424
    native kernel module intuition.library.kmod+0x00008064
    YAM:main()+0xba0 (section 1 @ 0x51C8)
    native kernel module newlib.library.kmod+0x0000208c
    native kernel module newlib.library.kmod+0x00002d34
    native kernel module newlib.library.kmod+0x00002eb0
    YAM:_start()+0x170 (section 1 @ 0x170)
    native kernel module dos.library.kmod+0x00022854
    native kernel module kernel+0x00041748
    native kernel module kernel+0x000417c8
    

PPC disassembly:
 0141c550: 7c842378   mr                r4,r4
 0141c554: 44000002   sc                
*0141c558: 4e800020   blr               
 0141c55c: 7c0802a6   mflr              r0
 0141c560: 9421fff0   stwu              r1,-16(r1)
```

## Notes

**Comment by tboeckel** [comment]
**Originally on 2012-07-11 16:23:45 +0200**

___
Please provide a MAIL debug log created by the debug version of the next nightly build. Details about the debug version can be found in the [FAQ](http://yam.ch/wiki/FAQ/Debugging#BasicsaboutYAMsdebuggingfacilities).

And please provide the log as an attachment instead of posting it as a comment.

**Comment by tboeckel** [comment]
**Originally on 2012-07-11 21:44:14 +0200**

___
No need for a debug log. I found the bug.

**Comment by tboeckel** [comment]
**Originally on 2012-07-11 21:44:28 +0200**

___
In (19b5737):
 * YAM_UT.c: fixed the dynamic string buffer functions to not use realloc() as this does not zero the additional space. This closes #332. Sorry Jens, realloc() was a nice idea, but it sucks!
